mxcube3/ui/actions/general.js

```diff
@@ -18,22 +18,48 @@ export function showErrorPanel(show, message = '') {
 
 export function getInitialStatus() {
   return function (dispatch) {
-    fetch('mxcube/api/v0.1/initialstatus', {
+    let state = {};
+
+    let motors = fetch('mxcube/api/v0.1/diffractometer/movables/state', {
+      method: 'GET',
+      credentials: 'include',
+      headers: {
+        'Accept': 'application/json',
+        'Content-type': 'application/json'
+      }
+    });
+    let beamInfo = fetch('mxcube/api/v0.1/beam/info', {
+      method: 'GET',
+      credentials: 'include',
+      headers: {
+        'Accept': 'application/json',
+        'Content-type': 'application/json'
+      }
+    });
+    let sampleVideoInfo = fetch('mxcube/api/v0.1/sampleview/camera/info', {
       method: 'GET',
       credentials: 'include',
       headers: {
         'Accept': 'application/json',
         'Content-type': 'application/json'
       }
-    }).then(function (response) {
-      if (response.status >= 400) {
-        throw new Error('Server refused to send initialstatus');
+    });
+    let diffractometerInfo = fetch('mxcube/api/v0.1/diffractometer/info', {
+      method: 'GET',
+      credentials: 'include',
+      headers: {
+        'Accept': 'application/json',
+        'Content-type': 'application/json'
       }
-      return response.json();
-    })
-    .then(function (json) {
-      dispatch(setInitialStatus(json));
     });
 
+    motors.then(response => { state.Motors = response.json() });
+    beamInfo.then(response => { state.beamInfo = response.json() });
+    sampleVideoInfo.then(response => { state.Camera = response.json() });
+    diffractometerInfo.then(response => { Object.assign(state, response.json()) });
+    
+    Promise.all([motors,beamInfo,sampleVideoInfo,diffractometerInfo]).then(() => {
```

Comment:
I see what you are trying to do here but it seems that Promise.all only cares about the promises on the first level. 

I played around a bit and I think this might be what you want:


export function getInitialStatus() {
  return function (dispatch) {

    let motors = fetch('mxcube/api/v0.1/diffractometer/movables/state', {
      method: 'GET',
      credentials: 'include',
      headers: {
        'Accept': 'application/json',
        'Content-type': 'application/json'
      }
    });
    let beamInfo = fetch('mxcube/api/v0.1/beam/info', {
      method: 'GET',
      credentials: 'include',
      headers: {
        'Accept': 'application/json',
        'Content-type': 'application/json'
      }
    });
    let sampleVideoInfo = fetch('mxcube/api/v0.1/sampleview/camera/info', {
      method: 'GET',
      credentials: 'include',
      headers: {
        'Accept': 'application/json',
        'Content-type': 'application/json'
      }
    });
    let diffractometerInfo = fetch('mxcube/api/v0.1/diffractometer/info', {
      method: 'GET',
      credentials: 'include',
      headers: {
        'Accept': 'application/json',
        'Content-type': 'application/json'
      }
    });
    
    Promise.all([motors, beamInfo, sampleVideoInfo, diffractometerInfo]).then((responses) => {
      Promise.all(responses.map((response) => response.json())).then((values) => {
        dispatch(setInitialStatus(Object.assign({}, ...values)));
      });
    });
  };
}

This will first wait for all http promises then it wait for the json promises. There is one modification that needs to be done on the server for this to work and that is to wrap all motors in a Motors object first and the same for the rest.